### PR TITLE
Add: Task6 - Custom ArrayList and HashSet

### PR DESCRIPTION
**Pull request by aliunina**

### CODE REVIEW

File: Task2/src/model/storage/TicketArrayList.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package model.storage;
+
+import model.ticket.Ticket;
+
+import java.util.Arrays;
+
+public class TicketArrayList {
+    private Ticket[] list;
+    private int size;
```
Comment (Darya-Drutsko) [inline Task2/src/model/storage/TicketArrayList.java]: Does the size differ from list.size ?)

File: Task2/src/model/storage/TicketArrayList.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        size = 0;
+    }
+
+    public void add(Ticket ticket) {
+        size++;
+        list = Arrays.copyOf(list, size);
+        list[list.length - 1] = ticket;
+    }
+
+    public Ticket getByIndex(int index) throws ArrayIndexOutOfBoundsException {
```
Comment (Darya-Drutsko) [inline Task2/src/model/storage/TicketArrayList.java]: What is the type of ArrayIndexOutOfBoundsException?

File: Task2/src/model/storage/TicketHashSet.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import model.ticket.Ticket;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+public class TicketHashSet implements Iterable {
+    static final Object DUMMY = new Object();
+    private HashMap<Ticket, Object> map;
```
Comment (Darya-Drutsko) [inline Task2/src/model/storage/TicketHashSet.java]: Please don't use here HashMap, use array or LinkedList or ArrayList instead

File: Task2/src/model/storage/CustomHashSet.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+                list.remove(i);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public boolean contains(T element) {
+        for (HashValue hashVal : list) {
+            if (hashVal.hashCode == element.hashCode()) {
```
Comment (Darya-Drutsko) [inline Task2/src/model/storage/CustomHashSet.java]: why only hashCode is checked? Collision could be here

File: Task2/src/model/storage/CustomHashSet.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public Iterator<T> iterator() {
+        return new Iterator<T>() {
```
Comment (Darya-Drutsko) [inline Task2/src/model/storage/CustomHashSet.java:57]: how this construction is called?

Comment (aliunina) [reply Task2/src/model/storage/CustomHashSet.java:57]: It is called anonymous inner class

File: Task2/src/model/storage/CustomHashSet.java (hunk 4 above)
Comment (aliunina) [reply Task2/src/model/storage/CustomHashSet.java]: fixed

**Review by Darya-Drutsko — APPROVED** [review]

**Pull request closed without merging**